Sources/NIOSingletonsPosix/PosixSingletons.swift

```diff
@@ -0,0 +1,61 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+@_exported import NIOSingletonResources
+import NIOPosix
+import NIOCore
+
+extension NIOSingletons {
+    /// A globally shared, lazily intialized EventLoopGroup  that uses `epll` as the selector mechanism.
+    public static var multiThreadedPosixEventLoopGroup: MultiThreadedEventLoopGroup {
+        return globalMultiThreadedPosixEventLoopGroup
+    }
+
+    /// A globally shared, lazily intialized EventLoopGroup with just one thread that uses `epoll` as the selector mechanism.
+    public static var singleThreadedPosixEventLoopGroup: MultiThreadedEventLoopGroup {
+        return globalSingleThreadedPosixEventLoopGroup
+    }
+
+    /// A globally shared, lazily intialized EventLoop that uses `epoll` as the selector mechanism. Backed by `NIOSingletons.singleThreadedPosixEventLoopGroup`.
+    public static var posixEventLoop: EventLoop {
+        return Self.singleThreadedPosixEventLoopGroup.next()
+    }
+
+    /// A globally shared, lazily intialized `NIOThreadPool` that can be used for blocking I/O and other blocking operations.
+    public static var posixBlockingPool: NIOThreadPool {
+        return globalPosixBlockingPool
+    }
+}
+
+private let globalMultiThreadedPosixEventLoopGroup: MultiThreadedEventLoopGroup = {
+    let group = MultiThreadedEventLoopGroup(_canBeShutDown: false,
+                                            numberOfThreads: NIOSingletons.suggestedMultiThreadedEventLoopGroupThreadCount)
+    _ = Unmanaged.passUnretained(group).retain() // Never gonna give you up,
+    return group
+}()
+
+private let globalSingleThreadedPosixEventLoopGroup: MultiThreadedEventLoopGroup = {
+    let group = MultiThreadedEventLoopGroup(_canBeShutDown: false, numberOfThreads: 1)
+    _ = Unmanaged.passUnretained(group).retain() // never gonna let you down.
+    return group
+}()
+
+private let globalPosixBlockingPool: NIOThreadPool = {
+    let pool = NIOThreadPool(numberOfThreads: NIOSingletons.suggestedBlockingPoolThreadCount)
+    pool.start()
+    _ = Unmanaged.passUnretained(pool).retain() // Make sure this is never deallocated (not strictly necessary).
```

Comment:
Why do we do this if it's held by a global?

Reply:
I'm not 100% sure if there's an actual language guarantee to not dealloc globals. Maybe after `main` or maybe if the compiler develops some ultra smarts to figure out you're not using it anymore.

Yes, not today but I thought let's better be safe than sorry, this isn't gonna get deinit'ed.